aiden-recommender/aiden_recommender/scrapers/field_extractors.py

```diff
@@ -0,0 +1,127 @@
+from pydantic import BaseModel
+from abc import ABC, abstractmethod
+import jmespath
+from datetime import datetime
+from pydantic_core import ValidationError
+from loguru import logger
+
+
+class AbstractFieldExtractor(ABC):
+    @staticmethod
+    def select(data: dict, query: str) -> any:
+        if not query:
+            return data
+        return jmespath.search(query, data)
+
+    @abstractmethod
+    def _extract(self, data: dict) -> any:
+        pass
+
+    @abstractmethod
+    def extract(self, data: dict, field: str) -> dict:
+        pass
+
+
+class FieldExtractorBase(AbstractFieldExtractor):
```

Comment:
interesting way to use pydantic :0

Reply:
Maybe there is a more pydantic way to do this actually, didn't think about it